hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieFileInputFormatBase.java

```diff
@@ -187,13 +286,28 @@ private static FileStatus getFileStatusUnchecked(Option<HoodieBaseFile> baseFile
               .map(fileSlice -> {
                 Option<HoodieBaseFile> baseFileOpt = fileSlice.getBaseFile();
                 Option<HoodieLogFile> latestLogFileOpt = fileSlice.getLatestLogFile();
-                if (baseFileOpt.isPresent()) {
-                  return getFileStatusUnchecked(baseFileOpt);
-                } else if (includeLogFilesForSnapShotView() && latestLogFileOpt.isPresent()) {
-                  return createRealtimeFileStatusUnchecked(latestLogFileOpt.get(), fileSlice.getLogFiles());
+                Stream<HoodieLogFile> logFiles = fileSlice.getLogFiles();
+
+                Option<HoodieInstant> latestCompletedInstantOpt =
+                    fromScala(fileIndex.latestCompletedInstant());
+
+                // Check if we're reading a MOR table
+                if (includeLogFilesForSnapshotView()) {
+                  if (baseFileOpt.isPresent()) {
+                    return createRealtimeFileStatusUnchecked(baseFileOpt.get(), logFiles, latestCompletedInstantOpt, tableMetaClient);
+                  } else if (latestLogFileOpt.isPresent()) {
+                    return createRealtimeFileStatusUnchecked(latestLogFileOpt.get(), logFiles, latestCompletedInstantOpt, tableMetaClient);
+                  } else {
+                    throw new IllegalStateException("Invalid state: either base-file or log-file has to be present");
+                  }
                 } else {
-                  throw new IllegalStateException("Invalid state: either base-file or log-file should be present");
+                  if (baseFileOpt.isPresent()) {
+                    return getFileStatusUnchecked(baseFileOpt.get());
+                  } else {
+                    throw new IllegalStateException("Invalid state: base-file has to be present");
+                  }
                 }
```

Comment:
It looks like the logic is changed here to cover broader cases, especially when both base file and log files exist.  Basically, if base file exists, log files are ignored before.  However, for Snapshot mode, should log files be ignored by default?  Or do you intend to use this method `listStatusForSnapshotMode()` broadly for different query mode (renaming it if true)?

Reply:
To clarify, for Snapshot query on MOR table, we do need the merging of base file and log files.  For Snapshot query on COW table, only base file is needed.  I think that is your intention of changes here.

Reply:
`listStatusForSnapshotMode()` is used for both COW/MOR, and whether we're handling COW/MOR is controlled by `includeLogFilesForSnapshotView` (will clean this up in a follow up, cleaning up some assertions after majority of the PRs land)